### PR TITLE
Add TCP and encryption support to plugtest.

### PR DESCRIPTION
**Pull request by boaks**
Prepare cf-plugtest server and client for eclipse sandbox.
Enable encryption support and experimental TCP feature.

Signed-off-by: Achim Kraus <achim.kraus@bosch-si.com>

### CODE REVIEW

**Comment by boaks** [comment]
Seems that I accidently formated the source. If someone thing it's import to go back to the old formating to see the changes easier, I would do it. Otherwise I think the proper format could be kept.
For coap and coaps the sandbox "californium.eclipse.org" is already up and running :-).

File: demo-apps/cf-plugtest-server/src/main/java/org/eclipse/californium/plugtests/PlugtestServer.java [hunk 1]
```diff
@@ -46,92 +60,203 @@
...
+			PlugtestServer server = new PlugtestServer();
+			// ETSI Plugtest environment
+			// server.addEndpoint(new CoAPEndpoint(new InetSocketAddress("::1",
+			// port)));
+			// server.addEndpoint(new CoAPEndpoint(new
+			// InetSocketAddress("127.0.0.1", port)));
+			// server.addEndpoint(new CoAPEndpoint(new
+			// InetSocketAddress("2a01:c911:0:2010::10", port)));
+			// server.addEndpoint(new CoAPEndpoint(new
+			// InetSocketAddress("10.200.1.2", port)));
```
Comment (sbernard31) [inline demo-apps/cf-plugtest-server/src/main/java/org/eclipse/californium/plugtests/PlugtestServer.java]: Comment formating is a bit strange here.
Should we keep thoses comments ?

Comment (boaks) [reply demo-apps/cf-plugtest-server/src/main/java/org/eclipse/californium/plugtests/PlugtestServer.java]: @mkovatsc 

Do you want to keep it, or could it be removed?

**Comment by boaks** [comment]
> its a bit strange that TCPConnector properties can be set through NetworkConfig as this is up to user to create its connector

It is also strange to have `COAP_SECURE_PORT` and `HTTP_PORT` (and even more `HTTP_???`).
So I don't think, this should be "cleaned up" in this PR :-).

**Comment by sbernard31** [comment]
> So I don't think, this should be "cleaned up" in this PR :-).

Totally agree.

**Comment by boaks** [comment]
If no objection, I would like to merge this.

**Comment by sbernard31** [comment]
👍

**Comment by boaks** [comment]
So, start the milestone?

**Comment by sbernard31** [comment]
yep :)
I do or you do ?

**Comment by boaks** [comment]
I try it,
(and you may sort it out, if it gets broken ...)

**Comment by boaks** [comment]
:-) 

Seems that the HIPP is "very stressed" today. We'll see, if it builds with the 3. try. 
Otherwise I retry it tomorrow.

**Pull request merged**